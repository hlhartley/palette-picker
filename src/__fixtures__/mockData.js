export const mockProjects = [
    {
        id: 1,
        name: "Warm colors project",
        created_at: "2019-03-26T05:17:02.689Z",
        updated_at: "2019-03-26T05:17:02.689Z"
    },
    {
        id: 2,
        name: "Cool colors project",
        created_at: "2019-03-26T05:17:02.702Z",
        updated_at: "2019-03-26T05:17:02.702Z"
    },
    {
        id: 3,
        name: "Neutral colors project",
        created_at: "2019-03-26T05:17:02.727Z",
        updated_at: "2019-03-26T05:17:02.727Z"
    }
]

export const mockPalettes = [
    {
        id: 1,
        project_id: 1,
        name: "Summer colors",
        color1: "FFFF82",
        color2: "DDDCF7",
        color3: "B5D99C",
        color4: "FFAF66",
        color5: "E65F5C",
        created_at: "2019-03-26T05:17:02.735Z",
        updated_at: "2019-03-26T05:17:02.735Z"
    },
    {
        id: 2,
        project_id: 1,
        name: "Spring colors",
        color1: "F25469",
        color2: "BD2D87",
        color3: "D664BE",
        color4: "DF99F0",
        color5: "B191FF",
        created_at: "2019-03-26T05:17:02.736Z",
        updated_at: "2019-03-26T05:17:02.736Z"
    },
    {
        id: 3,
        project_id: 2,
        name: "Fall colors",
        color1: "FCD0A1",
        color2: "B1B695",
        color3: "A690A4",
        color4: "EFA958",
        color5: "AFD2E9",
        created_at: "2019-03-26T05:17:02.751Z",
        updated_at: "2019-03-26T05:17:02.751Z"
    },
    {
        id: 4,
        project_id: 3,
        name: "Winter colors",
        color1: "594157",
        color2: "726DA8",
        color3: "7D8CC4",
        color4: "A0D2DB",
        color5: "BEE7E8",
        created_at: "2019-03-26T05:17:02.757Z",
        updated_at: "2019-03-26T05:17:02.757Z"
    }
]

export const mockHexcodes = [
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    }
]