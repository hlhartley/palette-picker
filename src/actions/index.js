export const setError = (error) => ({
  type: 'SET_ERROR', 
  error
});

export const setLoading = (isLoading) => ({
  type: 'SET_LOADING', 
  isLoading
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS', 
  projects
});

export const setPalettes = (palettes) => ({
  type: 'SET_PALETTES', 
  palettes
});

export const setCurrentPaletteId = (paletteId) => ({
  type: 'SET_CURRENT_PALETTE_ID', 
  paletteId
});

export const setCurrentProjectName = (projectName) => ({
  type: 'SET_CURRENT_PROJECT_NAME', 
  projectName
});

export const setRandomHexcodes = (hexcodes) => ({
  type: 'SET_RANDOM_HEXCODES', 
  hexcodes
});

export const toggleDisplayRandom = (shouldDisplay) => ({
  type: 'DISPLAY_RANDOM', 
  shouldDisplay
});
