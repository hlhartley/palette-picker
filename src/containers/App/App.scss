* {
  margin: 0;
  padding: 0;
}

html, body {
  height: 100%;
}

.app-content {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  height: 90vh;
  background-color: #f5f5fb;
}