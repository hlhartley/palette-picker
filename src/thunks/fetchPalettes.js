import { setLoading, setError, setPalettes } from '../actions';
import { fetchData } from '../utils/api.js';

export const fetchPalettes = (id) => {
  let endpoint = `/projects/${id}/palettes`;
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const result = await fetchData(endpoint, 'GET');
      dispatch(setPalettes(result));
      dispatch(setLoading(false));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}