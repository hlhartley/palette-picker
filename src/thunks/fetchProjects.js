import { setError, setProjects } from '../actions';
import { fetchData } from '../utils/api.js';

export const fetchProjects = () => {
  const endpoint = '/projects';
  return async (dispatch) => {
    try {
      const result = await fetchData(endpoint, 'GET');
      dispatch(setProjects(result));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}