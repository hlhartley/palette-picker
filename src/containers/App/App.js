import React, { Component } from 'react';
import { fetchProjects } from '../../thunks/fetchProjects';
import { fetchPalettes } from '../../thunks/fetchPalettes';
import { connect } from 'react-redux';
import Header from '../../components/Header/Header';
import ParentSection from '../../components/ParentSection/ParentSection';

export class App extends Component {
  constructor() {
    super();
    this.state = {
    }
  }

  componentDidMount = () => {
    this.initialFetch();
  }

  initialFetch = async () => {
    await this.props.fetchProjects();
    await this.props.projects.forEach((project) => {
      this.props.fetchPalettes(project.id);
    })
  }

  render() {
    return (
      <div className="App">
				{!this.props.isLoading ? (
        <header className="App-header">
          <Header />
          <ParentSection />
        </header>
        ) : (
          <p>Loading...</p>
        )}
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  projects: state.projects,
  palettes: state.palettes,
  currentPaletteId: state.currentPaletteId,
  isLoading: state.isLoading,
  error: state.error
});

export const mapDispatchToProps = (dispatch) => ({
  fetchProjects: () => dispatch(fetchProjects()),
  fetchPalettes: (id) => dispatch(fetchPalettes(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
