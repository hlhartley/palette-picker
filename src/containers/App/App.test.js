import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { fetchProjects } from '../../thunks/fetchProjects';
import { fetchPalettes } from '../../thunks/fetchPalettes';
import { mockProjects } from '../../__fixtures__/mockData';
import { setLoading } from '../../actions';

jest.mock('../../thunks/fetchProjects')
jest.mock('../../thunks/fetchPalettes')

describe('App', () => {
  let wrapper;
  let fetchProjectsMock = jest.fn();
  let fetchPalettesMock = jest.fn();
  let setLoadingMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <App 
        fetchProjects={fetchProjectsMock}
        fetchPalettes={fetchPalettesMock}
        setLoading={setLoadingMock}
        projects={mockProjects}
      />
      )
    });
  describe('App component', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('Should call initialFetch on mount', () => {
      wrapper.instance().initialFetch = jest.fn();
      wrapper.update();
      wrapper.instance().componentDidMount();
      expect(wrapper.instance().initialFetch).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with projects and isLoading as keys', () => {
      const mockState = {
        projects: [],
        isLoading: true,
        error: '',
        palettes: [],
        extra: ''
      }
      const expected = {
        projects: [],
        isLoading: true
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  }); 

  describe('mapDispatchToProps', () => {
    it('should call dispatch when fetchProjects is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchProjects();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchProjects()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch when fetchPalettes is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchPalettes();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchPalettes()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch when setLoading is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = setLoading();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setLoading()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});
