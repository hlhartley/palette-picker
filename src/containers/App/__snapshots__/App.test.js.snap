// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should properly render the component elements 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <Header />
    <ParentSection />
  </header>
</div>
`;
